pkg/elemental/elemental.go

```diff
@@ -585,11 +542,17 @@ func DeployRecoverySystem(cfg types.Config, img *types.Image) error {
 	return nil
 }
 
-// DumpSource sets the image data according to the image source type
-func DumpSource(c types.Config, target string, imgSrc *types.ImageSource) error { // nolint:gocyclo
+// DumpSource dumps the imgSrc data to target. Mirror arguments sets the perfectly mirror source to target regardless if
+// some data is already present in target
+func DumpSource(c types.Config, target string, imgSrc *types.ImageSource, mirror bool) error { // nolint:gocyclo
 	var err error
 	var digest string
 
+	syncFunc := utils.SyncData
```

Comment:
Since DumpSource is only used from build-iso and MirrorData method I would even suggest to take syncFunc as a parameter. Although it feels kinda weird since it is not used if the source is a container image...

Reply:
In fact this was my first attempt, but since the signature of this method is big I though this was probably too verbose to put it as a parameter. But you got a point, probably should be set as a parameter and just hint in method comment this is most likely for `utils.SyncData` or `utils.MirrorData`.

Reply:
@frelon moved it as a parameter, makes the code a bit harder to read but more flexible and without adding extra logic.